rust/src/jsonbuilder.rs

```diff
@@ -602,17 +646,27 @@ impl JsonBuilder {
     ///
     /// The string is encoded into an intermediate vector as its faster
     /// than building onto the buffer.
+    ///
+    /// TODO: Revisit this, would be nice to build directly onto the
+    ///    existing buffer.
```

Comment:
I think this is worth revisiting this part.
Note that std::str::from_utf8 error case seems unreachable (as you encode the string correctly directly), so you should be able to push onto the self.buf directly